Source/cmake/WebKitCommon.cmake

```diff
@@ -258,6 +261,67 @@ if (NOT HAS_RUN_WEBKIT_COMMON)
                     "${CMAKE_SOURCE_DIR}"
                     "${CMAKE_BINARY_DIR}"
         )
+    elseif (CMAKE_EXPORT_COMPILE_COMMANDS)
+        # Create a simple symlink in DeveloperTools/compile_commands.json so that
+        # it can be relied upon regardless of unified or non-unified builds.
+        add_custom_command(
+            OUTPUT ${CMAKE_BINARY_DIR}/DeveloperTools/compile_commands.json
```

Comment:
We could, but when using unified builds it would still be generated at build time, so I see no reason to do things differently for non-unified builds.

Ideally I'd like both to be done at CMake generation time. Unfortunately CMake doesn't support that. I looked even into the CMake source code to look for somewhere to hook it into, but `compile_commands.json` is only written at the end of generation, long after all configure scripts have run.